### ISSUE TITLE
Fastboot Question

### ISSUE THREAD

**Issue by andysmithdfc** [Thinstation/thinstation#523]
Hello all,

I'm hoping someone can help me with regards to fastboot on the latest version of Thinstation. 

I have a working setup on an Ubuntu box, picking up the relevant initrd, vmlinuz etc files via TFTP from a root directory.
As soon as I set 'param fastboot true' in my build.conf file, and rebuild, I get the error 
'/lib is not a mountpoint - Fastboot was enabled but something went wrong with the transfer or mount'
I have placed the lib.squash file in the TFTP root directory with no luck.

Any help/advice would be much appreciated!

Andy

**Comment by Doncuppjr** [comment]
You might need autonet instead of network manager.

**Comment by andysmithdfc** [comment]
Thanks for replying! I couldn't see an autonet package in build.conf, I did see a ts-classic package which I read somewhere is the same thing? I have tried both packages without success.

**Comment by Thinstation** [comment]
Not the same, just pulls it in along with other classic packages. Just
comment out network manager and type package autonet, then build.

**Comment by andysmithdfc** [comment]
I have tried adding the autonet package in build.conf this morning, then rebuilding, with no success unfortunately.
Any other ideas?

**Comment by Doncuppjr** [comment]
You should be able to do some diagnostics from the shell. Can you download lib.squash manually? Does it fit in ram?

**Comment by andysmithdfc** [comment]
The file does fit it RAM...I am having trouble downloading the file to the thin client manually, but I am working at this..

Is there a way to change the default mount location of the lib.squash file, from /lib to another directory?

**Comment by Doncuppjr** [comment]
?? You can mount it anywhere you want, but it needs to be mounted at /lib to be functional.

**Comment by Lantizia** [comment]
He's not wrong you know, I get the same error.  

It loads in vmlinuz and initrd just fine before then simply saying...
```
/lib is not a mountpoint
Fastboot was enabled, but something went wrong with the transfer or mount.

Dropping to shell.
```
To ensure this was a clean build... I used a freshly installed new desktop to run the below commands... (and that desktop runs Ubuntu MATE 18.04, but that should be immaterial)...

```
git clone --depth 1 git://github.com/Thinstation/thinstation.git
cd thinstation/
./setup-chroot
cd /build
./build
exit
dnsmasq -i enp2s0 -F192.168.1.100,192.168.1.200 -Mboot/pxelinux/pxelinux.0 --enable-tftp --tftp-root="$(pwd)/build/boot-images/pxe" -d -p0 -K --log-dhcp --bootp-dynamic
```

I don't even see in my dnsmasq logs that it's even trying to find a lib.squash via TFTP.

Adding in "package autonet" and commenting out "package networkmanager" didn't change the error.

**Comment by andysmithdfc** [comment]
Yes, I never managed to get to the bottom of this one either. Always came back with the same error.

**Comment by Doncuppjr** [comment]
Perhaps it’s trying to get it via http. On the failing terminal, cat /proc/cmdline

**Comment by Lantizia** [comment]
![image](https://user-images.githubusercontent.com/10448369/73828120-9c6be680-47f8-11ea-914b-dd199c8f4040.png)

`usbhid.quirks=0x1bfd:0x3050:0x40 vt.global_cursor_default=0 loglevel=3 systemd.show_status=false udev.log_priority=3 rd.systemd.show_status=false rd.udev.log_priority=3 i915.fastboot=1 BOOT_IMAGE=::/boot/vmlinuz initrd=::/boot/initrd  splash=silent,theme:default console=tty1 LM-3`

**Comment by Doncuppjr** [comment]
Try manually transferring the file to /tmp, then mount at /lib

**Comment by Lantizia** [comment]
Hmm so for a moment I thought it might be that the network wasn't up.  But after making sure I did an "all modules build" and using autonet... sure enough the network was working.  But!  Same error.

Using the busy box tftp command though I can transfer the image to /tmp and mount it properly.

So still not sure :s

**Comment by Lantizia** [comment]
Weirdly if I use 6.2.2 (from the Releases tab) and don't clone from git (which I assume gives 6.2.3 - although I can't find any zip/tar.gz for that) then I instead get "There are no autostarted sessions!".

**Comment by Lantizia** [comment]
.

**Comment by Lantizia** [comment]
Using the BusyBox tftp command I can transfer the image to /tmp and mount it properly.

Is there a way to see the actual location it is attempting to get the file from?  I know it *should* be that it just looks using the same method and path as it used to get vmlinuz & initrd - but as I said, no such attempts are made via TFTP get it.

FASTBOOT_URL is *not* set overriding this.

Could it possibly be that the network isn't quite ready at the moment it makes the attempt (but is fine by the time I type in "ip addr show") ??  If so is there a way of debugging this or maybe adding something to pause it for a few seconds!?

Everything seems to work with fastboot set to false by the way :S

It would also be nice to see a 6.2.3 zip/tar.gz file as it seems to have been officially released as a stable version and there is a devstation ISO available for it over on SourceForge.

Thanks

**Comment by Doncuppjr** [comment]
It looks like busybox changed it's DHCP request packet..... once upon a time, busybox would ask for the 'next' server in it's request. This was set with option 66. Most desktop dhcp clients don't ask for the next server. They will accept a 'tftp' server if it is defined. Soooo, on your DHCP server make sure are assigning option 150 the same value as the 'next' server.






On Wednesday, February 5, 2020, 07:09:04 AM MST, Lantizia <notifications@github.com> wrote: 





Using the BusyBox tftp command I can transfer the image to /tmp and mount it properly.

Is there a way to see the actual location it is attempting to get the file from? I know it should be that it just looks using the same method and path as it used to get vmlinuz & initrd - but as I said, no such attempts are made via TFTP get it.

FASTBOOT_URL is not set overriding this.

Could it possibly be that the network isn't quite ready at the moment it makes the attempt (but is fine by the time I type in "ip addr show") ?? If so is there a way of debugging this or maybe adding something to pause it for a few seconds!?

Everything seems to work with fastboot set to false by the way :S

It would also be nice to see a 6.2.3 zip/tar.gz file as it seems to have been officially released as a stable version and there is a devstation ISO available for it over on SourceForge.

Thanks

**Comment by Thinstation** [comment]
Optionally, define 
NET_FILE_ALTERNATE=ip address of tftp server.

**Comment by Lantizia** [comment]
If I put **NET_FILE_ALTERNATE=192.168.1.91** in to the buildtime config file and rebuild... then I can at least see an attempt to get lib.squash on the dnsmasq built in tftp server... but it looks for...

::/boot/lib.squash
**when it should be...**
/boot/lib.squash

Anyway to get rid of those :: ?

**Comment by Doncuppjr** [comment]
No. Those are appropriate. Perhaps you need to tell dnsmasq to chroot the root. Why pxelinux, it’s so slow. Switch to lpxelinux. I did verify that busy box dhcpc still accepts option 66 for the server IP, not sure what dnsmasq is handing out.

**Comment by Lantizia** [comment]
Yeah so I've since determined that it was me running dnsmasq in proxy mode that was causing some of the issues.  I've yet to find a way to use dnsmasq in proxy mode whilst also having it correctly tell clients what the right TFTP server IP should be :S

However even when I run it **not** in proxy mode it still tries to read the file with the :: in the path via TFTP... which of course isn't valid.  So that really is something to do with ThinStation as that's what is being requested!

EDIT: Look like this has been fixed this in commit... https://github.com/Thinstation/thinstation/commit/d4a8167dfc73203b5916a41d75c4b6247152236e

Not sure if this resolves the issue for @andysmithdfc though... presuming you're using autonet and have the correct modules loaded for the clients NIC... this should now work via pxelinux (although admittedly lpxelinux is a lot quicker via HTTP).
